R/model_comparison.R

```diff
@@ -6,19 +6,21 @@
 #' @return rhats table
 #' @examples
 #' data(chagas2012)
-#' data_test <- prepare_serodata(serodata = chagas2012)
-#' model_constant <- run_seromodel(serodata = data_test,
+#' serodata <- prepare_serodata(serodata = chagas2012)
+#' model_constant <- run_seromodel(serodata = serodata,
 #'                                 foi_model = "constant",
 #'                                 n_iters = 1500)
-#' get_table_rhats(seromodel_object = model_constant)
+#' get_table_rhats(seromodel_object = model_constant,
+#'                 serodata = serodata)
 #' @export
-get_table_rhats <- function(seromodel_object) {
-  rhats <- bayesplot::rhat(seromodel_object$seromodel_fit, "foi")
+get_table_rhats <- function(seromodel_object, serodata) {
+  rhats <- bayesplot::rhat(seromodel_object, "foi")
 
   if (any(is.nan(rhats))) {
     rhats[which(is.nan(rhats))] <- 0
   }
-  model_rhats <- data.frame(year = seromodel_object$exposure_years, rhat = rhats)
+  exposure_years <- (min(serodata$birth_year):serodata$tsur[1])[-1]
```

Comment:
This not completely straightforward operation is done in a couple of places in the code. Would it be helpful to define a `get_exposure_years()` internal helper?

Reply:
I defined instead `get_cohort_ages`. This returns a data.frame with the birth year of each cohort and its corresponding age at the time of the survey. The replacement goes like:
```
cohort_ages <- get_cohort_ages(serodata)
exposure_years <- cohort_ages$birth_year
exposure_ages <- cohort_ages$age
```